objcgen/driver.cs

```diff
@@ -152,6 +162,18 @@ public void SetPlatform (string platform)
 			case "mac":
 				Platform = Platform.macOS;
 				break;
+			case "macosmodern":
+			case "macos-modern":
+				Platform = Platform.macOSModern;
+				break;
+			case "macosfull":
+			case "macos-full":
+				Platform = Platform.macOSFull;
+				break;
+			case "macossystem":
+			case "macos-system":
+				Platform = Platform.macOSSystem;
+				break;
```

Comment:
This needs to be included in our `/docs/` documentation. A large part can likely just link to back the current XM documentation (for the differences between XM profiles). What's need to be clear is the difference/when XM is required.

Maybe a `Targets.md` document that list the choices and describe them ?